### PR TITLE
Milestone4: Setup controller and view

### PR DESCRIPTION
**Pull request by isume295**
In this milestone I:

- Setup controllers and views
- Used params from browser requests in a safe way
- Used controllers to handle requests and render empty views.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,4 +3,8 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  root 'home#index'
+  resources :users, only: [:show] do
+    resources :posts, only: [:index, :show]
+  end
```
Comment (KanzaTahreem) [inline config/routes.rb]: - [ ] Great job on working through the routes! 🥇🥇 I noticed that you've set a root route to `home#index`, which is fantastic for defining the landing page of your application. However, there's an additional route mentioned in the project requirements that's currently missing. That's what I'm currently getting 👇

image

That's what we should have according to the [project requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Setup_Controllers_project_v1_1.md)

image

To achieve this, you can add a route for listing all users like this 👇

``` ruby
# This is your root route, which sets the default landing page of your application
root 'home#index'

# This route displays a list of all users, as per the project requirements
get '/users', to: 'users#index'

# or you can simply use the following for users
resources :users, only: [:index, :show]
```

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class UsersController < ApplicationController
+  def show
+    @user_id = params[:id]
+    # Placeholder action for the 'users/:id' URL
+  end
+end
```
Comment (KanzaTahreem) [inline app/controllers/users_controller.rb]: - [ ] Likewise, please make sure to create an `index` action for `users_controller`. The `index` action typically displays a list of all users in your application. We need to create this action in the appropriate controller, as per the wireframes we have. I noticed that you have a `home_controller` with an `index` action. Kudos for implementing it 🎉✨ However, it's not a requirement. It's entirely your choice, and I genuinely appreciate that. But, in any case, please make sure to adhere to the project requirements.

File: app/views/users/show.html.erb
Comment (KanzaTahreem) [inline app/views/users/show.html.erb:1]: - [x] Please make sure that we have a corresponding view files for each of the controller actions. Currently, the view file corresponding to user's `index` action (`users/index.html.erb`) is missing ⚠ Please make sure to fix this in accordance with the above comment and as mentioned in the [project requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Setup_Controllers_project_v1_1.md)

image

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @isume295 

Good job so far 👏! I really appreciate the effort you put in!  **_However, there are few this that still need to be fixed. But you are almost there!_** 💪 🏁

so close


## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ Controllers are created
- ✅ Actions and view files are present
- ✅ Routes are present

## Required Changes 🛑

_Please check the comments under the review._

## Optional suggestions ❕

N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/show.html.erb
Comment (isume295) [reply app/views/users/show.html.erb:1]: done

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (isume295) [reply app/controllers/users_controller.rb]: done

File: config/routes.rb (hunk 1 above)
Comment (isume295) [reply config/routes.rb]: done

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @isume295👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Good job with controllers. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**